securedrop/tests/test_unit_journalist.py

```diff
@@ -208,18 +217,29 @@ def test_edit_hotp(self):
 
     # TODO: more tests for admin interface
 
+    def test_delete_source_deletes_submissions(self):
+        """Verify that when a source is deleted, the submissions that
+        correspond to them are also deleted."""
+
+        source, files = self.add_source_and_submissions()
+
+        journalist.delete_collection(source.filesystem_id)
+
+        # Source should be gone
+        results = db_session.query(Source).filter(Source.id == source.id).all()
+        self.assertEqual(results, [])
+
+        # Submissions should be gone
+        results = db_session.query(Submission.source_id == source.id).all()
+        self.assertEqual(results, [])
+
```

Comment:
Ah, good point - I have now written this in a separate unit test (and added a helper function to setup test replies in the database).